examples/domainmodel/src/cli/generator.ts

```diff
@@ -0,0 +1,84 @@
+/******************************************************************************
+ * Copyright 2021 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import fs from 'fs';
+import _ from 'lodash';
+import { CompositeGeneratorNode, IndentNode, NL, processGeneratorNode } from 'langium';
+import { AbstractElement, Domainmodel, Entity, Feature, isEntity, isPackageDeclaration, Type } from '../language-server/generated/ast';
+
+export class DomainModelGenerator {
```

Comment:
I'd propose to implement this as a plain function instead:
```typescript
export function generateJava(ctx: GeneratorContext) {
    // ...
}

export interface GeneratorContext {
    domainmodel: Domainmodel;
    destination: string;
    path: string;
}
```

Reason: Here the `class` construct is used to store the parameters for the generator. This is how you would do it in Java or other object-oriented languages, but the TypeScript / JavaScript world is different. I know we already use classes a lot in our code base, but that's for other reasons:
 * Combine one or more service functions in a service class
 * Enable fine-grained customization with template methods
So our usage of classes is constrained to _service implementations_. We compared this with possible alternatives that work with plain functions, but found the class-based code to be most readable in that case. But in cases where plain functions can be used, I'd prefer them over classes.

Since this is a fundamental decision, it would be good to discuss it more broadly. @langium/committers WDYT?

Reply:
I agree with Miro here: We should try to stay away from classes as much as possible.

Reply:
Sounds reasonable to me as well

Reply:
I say "Yes!" to that too.

Reply:
Okay, I'll replace this to a plain function (in other CLIs too)